code/src/components/product-card/ProductCard.test.tsx

```diff
@@ -18,17 +19,33 @@ const mockProduct: Product = {
   price: 500
 };
 
+const mockCartIconClickHandler = jest.fn();
+
+jest.mock("react", () => ({
+  ...jest.requireActual("react"),
+  useState: jest.fn(() => [true, () => {}])
+}));
+
+const mockSetState = jest.fn();
+
 describe("ProductCard component", () => {
   beforeEach(() => {
+    (useState as jest.Mock).mockImplementation((init) => [init, mockSetState]);
```

Comment:
if you remove this mock and just rewrite smth that it can be working I guess and `beforeEach` will be more cleared
```jsx
jest.mock("react", () => ({
  ...jest.requireActual("react"),
  useState: jest.fn(() => [true, mockSetState])
}))
```
If you pass `mockSetState` here. What do you think ?

Reply:
Yes, it looks better. Fixed